network/src/sync/memory_pool.rs

```diff
@@ -67,13 +67,13 @@ impl Node {
         source: SocketAddr,
         transaction: Vec<u8>,
     ) -> Result<()> {
-        if let Ok(tx) = Testnet1Transaction::read_le(&*transaction) {
-            let inserted = self.expect_sync().consensus.receive_transaction(tx.serialize()?).await;
+        let tx = Testnet1Transaction::read_le(&*transaction)?;
```

Comment:
this is going to error out when people send us invalid transactions -- is that the intention?

Reply:
yes, and it's acocounted for in the inbound handler and the failure for the peer gets registered; we wouldn't want anyone to be able to spam the node with lots of invalid transactions that don't even parse properly